HO adding all article files. (#149)
hassan-obeid/tr_b_causal_2020: article/sections/_3_graph_overview.tex

\section{Overview of Causal Graphs}
\label{sec:graph-overview}

> Causal diagrams and causal graphical models have been introduced by \citet{pearl_causality_2000} as a powerful tool for causal inference, especially in observational studies.

Reviewer: ```suggestion
Causal diagrams have been introduced by \citet{pearl_causality_2000} as a powerful tool for causal inference, especially in observational studies.
```

Reviewer: After I read lines 8-11, I thought that it would be best to start there by calling causal diagrams DAGs from the jump, instead of later.

> Perhaps one of the most important and useful features of causal
> graphs when dealing with causal inference problems is the clear illustration
> of the causal relationships between the variables.

Reviewer: ```suggestion
Perhaps one of the most important and useful features of causal diagrams is their clear illustration
of the causal relationships between the variables.
```

While a formal introduction to the topic of directed acyclic graphs (DAGs) is beyond the scope of this chapter,
here we focus specifically on illustrating the power of DAGs to represent
and encode complex  causal relationships between variables in an intuitive and

> clear manner.

Reviewer: ```suggestion
A formal introduction to the topic of directed acyclic graphs (DAGs) is beyond the scope of this chapter.
Instead, we focus on illustrating the power of DAGs in representing complex causal relationships between variables in an intuitive and clear manner.
```

> Interested readers can refer to \citet{pearl_causality_2000} for a thorough introduction.

Reviewer: ```suggestion
Interested readers can refer to \citet{pearl_causality_2000} for a thorough introduction on DAGs and the graphical causal inference framework.
```

Consider the causal graph represented in Figure \ref{fig:simple-graph}.

> Suppose we're interested in the effect of Z on Y.

Reviewer: ```suggestion
Suppose we are interested in the effect of the treatment variable Z on the outcome variable, Y.
```

What the graph in Figure \ref{fig:simple-graph} implies is that Z is
independent of Y(Z) given X, or in other words, the assignment mechanism is

> ignorable conditional on the values of X.

Reviewer: ```suggestion
What the graph in Figure \ref{fig:simple-graph} implies is that Z is
independent of Y(Z) given X. In other words, the assignment mechanism is
ignorable conditional on the values of X.
```

In such situations, it is sufficient to control for
X to obtain an unbiased estimate of the causal effect of Z on Y.

\begin{figure}[h!]
   \centering
   \includegraphics[height=0.5\textwidth]{simple-graph.pdf}
   \caption{Simple DAG of the causal relationships between X, Y and Z}
   \label{fig:simple-graph}
\end{figure}

In comparison, note the set of structural equations needed to convey the same assumptions as Figure \ref{fig:simple-graph}.
\[Z = f_Z(X, \epsilon_Z)  \]

\[Y(z) = f_Y(X, z, \epsilon_Y)  \]

Now consider the case where there exists another latent confounding variable, U, which also affects both the treatment Z, as well as the outcome, Y.
Figure \ref{fig:simple-graph-confounded} illustrates this assumption in DAG form, and the equations below are the structural equation equivalent:

\[Z = f_Z(X, U, \epsilon_Z)  \]

\[Y(z) = f_Y(X, U, z, \epsilon_Y)  \]

\begin{figure}[h!]
   \centering
   \includegraphics[height=0.5\textwidth]{simple-graph-confounded.pdf}
   \caption{Simple DAG of the causal relationships between X, Y, Z and confounder U}
   \label{fig:simple-graph-confounded}
\end{figure}

> It is clear from Figure \ref{fig:simple-graph-confounded} that conditioning on X alone does not block all the indirect paths between Z and Y, and that ommitting U from the analysis will yield biased results of the causal relationship between Z and Y.

Reviewer: ```suggestion
It is clear from Figure \ref{fig:simple-graph-confounded} that conditioning on X alone does not block all the indirect paths between Z and Y, and that ommitting U from the analysis will yield biased results of the causal effect of the treatment Z on the outcome, Y.
```

The structural equations also show that the ignorability assumption
of Z does not hold if we only condition on X, and we risk obtaining biased estimates of the causal effect of Z on Y if we fail to

> account for U, sometimes referred to as a common cause.

Reviewer: ```suggestion
of Z does not hold if we only condition on X. If we fail to account for the latent variable U, we risk obtaining biased estimates of the causal effect of Z on Y.
```

Even in this very simple example with only three or four variables, one can see the advantage of using a causal graph to encode assumptions.

> This advantage of DAGs becomes even more apparent in larger problems with many covariates available, and where the causal structure of the data is way more complicated, which are the types of problems typically faced by transportation demand modelers.

Reviewer: When I read this segment, I understood that only travel demand modelers face this example of complex problems. Can we find a better way of communicating that the majority of travel demand modeling problems are that complex without sounding like they are the only ones that complex?

```suggestion
Even in these very simple examples with only three or four variables, one can start noticing the advantage of using a causal graph to encode modeling assumptions.
This advantage of DAGs becomes even more apparent in problems with many covariates and a more complex causal structure of the data.
```

Another benefit of DAGs is that they come with a set of testable implications, and incorporating them in any causal

> analysis adds robustness and defensibility to one's analysis.

Reviewer: Do you mean testing for these implications?

> We discuss those implications in section 5, and illustrate how to use those tests in one's analysis.

Reviewer: ```suggestion
We discuss these implications and detail how to test for them in in section 5.
```

> Lastly, it is important to note that the graphical approach to causality focuses
> primarily on issues of identification of causal effects, that is, given a
> directed acyclic graph (DAG) that encodes an analyst's knowledge and belief
> about the data generation process of the problem at hand, can a specific
> causal effect be identified?

Reviewer: ```suggestion
Lastly, it is important to note that the graphical approach to causality focuses
primarily on issues of identification of causal effects. That is whether a specific
causal effect can be identified given a DAG that encodes an analyst's knowledge and belief about the data generation process of the problem at hand.
```

As such, we emphasize that DAGs are great tools
for a modeler to encode their assumptions about a problem, but not necessarily
a guide on how to estimate a causal effects of interest.

\subsection{Prior Uses of Causal Graphs in Choice Modelling}
\label{sec:choice-graphs}

In our last subsection, we reviewed the basics of causal graphs: what are they and why are they useful?
We targeted that subsection at choice modellers who are unfamiliar with these tools.
However, readers should be aware of the history of causal graphs in choice modelling.
Indeed, choice modellers have used causal graphs (in limited fashion) for years.

> Here are three examples to illustrate our point.

Reviewer: My suggestion here is to prime them for these three different types of models then present them, aka tell them what you'll tell them about, then tell them about it.

First, consider the case of Random Utility Maximization (RUM) models.

\begin{figure}
   \centering
   \includegraphics[width=0.25\textwidth]{rum-causal-graph}
   \caption{Archetypical RUM causal diagram}
   \label{fig:example-graph-rum}
\end{figure}

For decades, choice modellers have used stylized DAGs to depict RUM models.
In particular, these diagrams illustrate an assumed choice-making (i.e., causal) process.
As an example, see Figure \ref{fig:example-graph-rum}, reproduced from Figure 1 of \citet{ben_2002_integration}.
\citeauthor{ben_2002_integration} draw a two-part process.
First, they assume that explanatory variables cause unobserved utilities.

> Then, they assume that the unobserved utilities cause the choice.

Reviewer: This is confusing to me..not sure how covariates cause unobserved utilities.

Note that although RUM diagrams adequately show the choice process, we still call them stylized.
Specifically, these diagrams lack detail about the relationships between explanatory variables.
When speaking collectively, we cannot tell if one explanatory variable causes another.
Unfortunately, as shown in Section \ref{sec:graph-importance}, such knowledge is crucial.
Without more detailed causal knowledge, our inferences may be inconsistent and arbitrarily bad.

\begin{figure}
   \centering
   \includegraphics[width=0.5\textwidth]{iclv-causal-graph}
   \caption{Archetypical ICLV causal diagram}
   \label{fig:example-graph-iclv}
\end{figure}

Besides RUM models, causal graphs appear in the literature on Integrated Choice and Latent Variable (ICLV) models.
An example of an ICLV model is in Figure \ref{fig:example-graph-iclv}, based on Figure 5 of \citet{ben_2002_integration}.
As with RUM models, ICLV causal graphs represent assumptions about the choice process.

> Here concern typically centers around unobserved mediators.
> These unobserved variables cause the outcome and are caused by one's observed covariates.

Reviewer: ```suggestion
In the case of ICLVs, the concern typically centers around unobserved mediators.
Unobserved mediators are variables that cause the outcome variable and are caused by one's observed covariates.
```

Omitting such mediators leads to models that misrepresent the assumed choice process.

> As a result, researchers care deeply about ICLV models that avoid these behavioral misrepresentations.

Reviewer: I'm confused about what this is the result of? The result of the misrepresentation of the choice process? Or something else?

\begin{figure}
   \centering
   \includegraphics[width=0.5\textwidth]{daysim-model-graph}
   \caption{Archetypical causal diagram for activity-based models}
   \label{fig:example-graph-abm}
\end{figure}

Finally, consider activity based travel demand models (ABM).

> These models often come with a causal diagram that depicts the interrelations between the outcomes in the model.

Reviewer: ```suggestion
These models often come with a causal diagram that depicts the interrelations between the outcome of each of the sub-models in the model.
```

E.g., household location choice, destination choice, travel mode choice, departure time choice, route choice etc.
For example, see Figure \ref{fig:example-graph-abm}, from \citet[Fig. 2]{bradley_2010_sacsim}.
The purpose of these graphs is to explain the structure of the entire system of outcome models.

In particular, ABM diagrams highlight two sets of researcher assumptions.

> They detail the researcher's beliefs about which choices, i.e. outcomes, precede others.

Reviewer: ```suggestion
First, they detail the researcher's beliefs about which choices, i.e. outcomes, precede others.
```

For instance, work location choice preceding auto-ownership choice.

> ABM diagrams also detail which downstream choices partially cause which preceding ones.

Reviewer: ```suggestion
Second, ABM diagrams also detail which choices partially cause particular preceding choices.
```

For example, considerations about one's travel mode choice influences one's daily activity choices.
This happens even though the activity choices precede the travel mode choices.

> Two main features distinguish RUM, ICLV, and ABM graphs from the causal graphs of \citet{pearl_1995_causal}.

Reviewer: ```suggestion
Two main features distinguish RUM, ICLV, and ABM graphs from DAGs introduced by \citet{pearl_1995_causal}.
```

> First, causal graphs in choice modelling traditionally ignore relations between the explanatory variables.

Reviewer: Wondering if we should say "choice modeling graphs" instead of "causal graphs".

```suggestion
First, causal graphs in choice modelling traditionally do not depict relationships between their explanatory variables.
```

Typically, choice modelling causal graphs show all explanatory variables together as a monolith.

> The relationships between explanatory variables is often not specified.

Reviewer: ```suggestion
```

Even worse, researchers may tacitly treat the variables as if they are jointly independent.
In the language used by a large group of causal inference scholars, econometric causal diagrams ignore the treatment assignment mechanism.

Secondly, causal graphs in choice modelling papers are purely didactic.
They convey how choice modellers perceive the world and the choice generation process.
However, they are seldom treated as a model themselves.

> In particular, choice modellers rarely test the predictions of causal graphs.

Reviewer: Implications or predictions?

In doing so, we underuse causal implications such as statistical independence.
Worse, we possibly violate these implications with great frequency.

> Such actions ignore the efforts from causal inference in computer science.
> There, researchers stress using their data to test the implications of their causal graphs.

Reviewer: ```suggestion
Such actions ignore the efforts from causal inference research in computer science.
Computer science causal inference researchers stress using the data to test the implications of any developed causal graphs.
```

In conclusion, choice modellers have long made use of causal graphs.
In select contexts, causal graphs convey causal assumptions about choice processes.

> Thus far, however, our field has underutilized these tools.
> We seldom use causal graphs to encode our assumptions about how our explanatory variables came to be.
> Moreover, we do not routinely test causal graphs against empirical choice data.

Reviewer: ```suggestion
Thus far, however, our field has underutilized causal graphs.
We seldom use causal graphs to encode our assumptions about how our explanatory variables came to be.
Moreover, we do not routinely test causal graphs against empirical choice data.
```

> These two issues are opportunities for choice modelling to gain from the insights and methods of causal inference scholars.
> The rest of the chapter will focus on the following three topics.
> How we can construct causal diagrams that pay attention to explanatory and outcome variables?
> How we can test a given causal graph against one's data?
> How can we deal with ``real-world'' graphs featuring unobserved confounding?

Reviewer: ```suggestion
These two issues are opportunities for choice modelers to gain from the insights and methods of causal inference scholars.
The rest of the chapter will focus on the following three topics:
 - Constructing causal diagrams that carefully consider the relationships between the different explanatory and outcome variables.
 - Testing a given causal graph against one's data.
 -  Dealing with ``real-world'' graphs featuring unobserved confounding?
```
